aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -978,30 +1089,45 @@ static void apply_svd(Tensor& self, Tensor& U, Tensor& S, Tensor& VT,
   int info;
   auto m = self.size(-2);
   auto n = self.size(-1);
-  auto k = std::min(m, n);
-  Tensor iwork = at::empty({8 * k}, at::kInt);
+  auto mn = std::min(m, n);
+  Tensor iwork = at::empty({8*mn}, at::kInt);
   auto iwork_data = iwork.data_ptr<int>();
+  Tensor rwork;
```

Comment:
@dylanbespalko why is this required? did you have tests to verify the svd behavior for complex?

Reply:
I am working to add svd for complex on cuda if we need to add something similar on cuda as well since magma follows lapack